src/get-started/java-binding.md

```diff
@@ -0,0 +1,382 @@
+```
+
+`Surefire/Failsafe` for Maven should be configured as follows:
+
+```xml
+<plugin>
+    <!-- You can also configure a failsafe to run integration tests during
+         'validate' phase of a Maven build to separate unit tests and ITs. -->
+    <artifactId>maven-surefire-plugin</artifactId>
+    <configuration>
+        <argLine>
+            -Djava.library.path=${path-to-java-bindings-library}
+        </argLine>
+    </configuration>
+</plugin>
+```
+
+Each test can create `Snapshot`, `Fork`, using `MemoryDb`, and also apply `Fork`
+to its state:
+
+??? note "MySchemaTest"
+    ```java
+    package com.exonum.binding.qaservice;
+
+    import com.exonum.binding.proxy.Cleaner;
+    import com.exonum.binding.storage.database.Fork;
+    import com.exonum.binding.storage.database.MemoryDb;
+    import com.exonum.binding.storage.database.Snapshot;
+    import org.junit.After;
+    import org.junit.Before;
+    import org.junit.Test;
+
+    public class MySchemaTest {
+
+      MemoryDb db;
+      Cleaner cleaner;
+
+      @Before
+      public void setUp() {
+        db = MemoryDb.newInstance();
+        cleaner = new Cleaner();
+      }
+
+      @After
+      public void tearDown() throws Exception {
+        cleaner.close();
+        db.close();
+      }
+
+      @Test
+      public void testCreatingSnapshot() {
+        Snapshot snapshot = db.createSnapshot(cleaner);
+        // Use the Snapshot to test the schema
+      }
+
+      @Test
+      public void testMergingSomeChanges() {
+        Fork fork = db.createFork(cleaner);
+        // Make some changes to Fork
+        // …
+
+        // Merge the changes made to the database Fork into the database
+        db.merge(fork);
+
+        // Create a Snapshot of the database state. Must include the changes
+        // made above.
+        Snapshot snapshot = db.createSnapshot(cleaner);
+        // Use the Snapshot to test the schema/transaction
+        // with a certain database state.
+        // …
+        }
+      }
+      ```
+
+##### Of Transactions
+
+To test transactions execution, you can use `MemoryDb`, as in the previous
+section.
+
+#### Of Read Requests
+
+To test read requests for service data, you can use a fake that implements
+`Node` interface and uses `MemoryDb` to create `Snapshot`:
+`com.exonum.binding.service.NodeFake`. The `MemoryDb` contents can be filled in
+by executing `MemoryDb#merge(Fork)` operation as in the section above.
+
+#### Of API
+
+To test API implemented by Vertx tools, use the tools described in the
+[project documentation](https://vertx.io/docs/vertx-unit/java/#_introduction)
+
+You can use [Vertx Web Client][vertx-web-client] as a client or another HTTP
+client.
+
+An example of API service tests can be found in
+`com.exonum.binding.cryptocurrency.ApiControllerTest`.
+
+### Known Limitations
+
+- Serialization is determined by a user, so Java services are not compatible
+  with JS light client (unless messages are updated).
```

Comment:
I believe "(unless messages are updated)" is an internal comment :upside_down_face: